### PR TITLE
Update for submission status

### PR DESCRIPTION
**Pull request by markpatton**
The submissions table, submissions detail page, and grants detail page have their logic for calculating submission status updated. In addition the submissions table will make a constant number of queries instead of being linear to the number of rows.

Due to limited time, there is code duplication in submissions/index and grants/detail routes as well as in the
submissions-status and submissions-status-cell components.

In order to test. Fire up the app login as various users and look at the submission status.

Note that I am not sure the logic is correct!

### CODE REVIEW

File: app/templates/components/submission-status.hbs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    {{#if (eq status "Submitted")}}
+      <span tooltip-position="left" tooltip="This submission has either been sent to all target repositories, or is in the queue and will be submitted shortly.">
+        <i class="fas fa-info-circle"></i> {{status}}
+      </span>
+    {{/if}}
+    {{#if (eq status "Stalled")}}
+      <span tooltip-position="left" tooltip="Manual intervention required.">
+        <i class="fas fa-info-circle"></i> {{status}}
+      </span>
+    {{/if}}
```
Comment (karenhanson) [inline app/templates/components/submission-status.hbs]: Per the description here, this should be `See details`, there is no stalled status show for a combined status calculations, so this should not be displayed on the Submissions table or at the top of Submission Details page

https://github.com/OA-PASS/pass-ember/issues/568

> * submission.source="pass" && submission.submitted=true && no repoCopy: display submitted
> * submission.source="other" && submission.submitted=false && no repoCopy : display Manuscript expected
> * ALL of repoCopy.copyStatus="complete": display complete
> * Other combination of repositoryCopy.copyStatus: display a link saying "See details" that links to the submission details page. This part relates to #539 which will have a nicer way of displaying all repositoryCopy.copyStatus

Comment (karenhanson) [reply app/templates/components/submission-status.hbs]: To clarify - the one at the top should say "see details" (is currently incorrect), the one on the repository should say "stalled" (is currently correct)

image

File: app/components/submission-status-cell.js [hunk 2]
```diff
@@ -1,43 +1,32 @@
...
-      if (source == 'pass' && submission.get('submitted')) {
-        return 'Submitted';
-      } else if (source == 'other' && !submission.get('submitted')) {
+    if (submission.get('source') == 'other' && !submission.get('submitted')) {
         return 'Manuscript expected';
+    }
+
+    if (repoCopies.get('length') < deposits.get('length')) {
+      if (submission.get('source') == 'pass') {
+        return 'Submitted';
```
Comment (karenhanson) [inline app/components/submission-status-cell.js]: Submitted should appear when there are zero RepositoryCopies for the Submission, `Submission.source=pass` and `Submission.submitted=true`. Currently these are appearing as more details links.

**Comment by markpatton** [comment]
The status logic really needs to be pulled out into a service where we can unit test the various permutations.

**Comment by markpatton** [comment]
I pushed the logic into a service and updated it as requested. I think. This needs a full specification and unit testing to really be confident about.

I won't have any time to work on this for a few days!

**Comment by htpvu** [comment]
I'll stew on this over the weekend.

**Comment by karenhanson** [comment]
I think because of a minor lag in creating the deposit and indexing, the query fails if you go right to the submission list from the Thank You page. There are supposed to be ~15 more submissions below this. Refreshing fixes it. So perhaps  the default "see details" can be shown if no records are found and `source=pass`?

image

File: app/services/submission-status.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    if (repoCopies.get('length') > 0 && repoCopies.get('length') == deposits.get('length')) {
+      if (repoCopies.every(rc => rc.get('copyStatus') == 'complete')) {
+        return 'Complete';
+      }
+    }
+
+    if (repoCopies.get('length') > deposits.get('length')) {
+      if (submission.get('source') == 'other' && submission.get('submitted')) {
+        return 'Complete';
+      }
```
Comment (karenhanson) [inline app/services/submission-status.js]: So this isn't exactly correct - you can have an in-progress submission added through NIHMS ETL. The repository Copy status in that instance would be "accepted" or "in-progress"... so in order to be complete you still need this logic: `if (repoCopies.every(rc => rc.get('copyStatus') == 'complete')) {`

**Comment by karenhanson** [comment]
Merged though there are still a few issues so we can deal with those on a separate PR

**Pull request merged**